### PR TITLE
Opensearch operator set connection type

### PR DESCRIPTION
**Pull request by Lukas1v**
---

We ran into issues when trying the Opensearch operator to connect to an endpoint with a self-signed certificate. It even failed with the following connection settings:
```
{
  "use_ssl": "True",
  "verify_certs": "False"
}
```
Those parameters were converted to strings and not as booleans.

Another issue we faced is that the connection type RequestsHttpConnection did not work for us. We discovered it because we knew the Opensearch client itself worked which uses by default a Urllib3HttpConnection, but the hook does not. 

This PR addresses both issues:
-  The parameters use_ssl and verify_certs are both converted to boolean in the hook
- an option is provided to specify an existing Opensearch connection type. If nothing is provided, RequestsHttpConnection is used as the default

**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/opensearch/hooks/opensearch.py [hunk 1]
```diff
@@ -40,15 +46,49 @@ class OpenSearchHook(BaseHook):
     conn_type = "opensearch"
     hook_name = "OpenSearch Hook"
 
-    def __init__(self, open_search_conn_id: str, log_query: bool, **kwargs: Any):
+    def __init__(
+        self,
+        open_search_conn_id: str,
+        log_query: bool,
+        open_search_conn_class: str = "RequestsHttpConnection",
```
Comment (Taragolis) [inline airflow/providers/opensearch/hooks/opensearch.py]: I don't think we need to use such complicated logic with dynamic imports, we could just provide class as is into the hook/operator

```python
from typing import Any, TYPE_CHECKING

if TYPE_CHECKING:
    from opensearchpy import Connection as OpenSearchConnectionClass

class OpenSearchHook(BaseHook):
    def __init__(
        ...
        open_search_conn_class: type[OpenSearchConnectionClass] | None = None,
```

Comment (Taragolis) [reply airflow/providers/opensearch/hooks/opensearch.py]: And also need add information about new attribute in docstrings

**Comment by Taragolis** [comment]
>It even failed with the following connection settings:
>
>{
>  "use_ssl": "True",
>  "verify_certs": "False"
>}

```json
{
    "use_ssl": true, 
    "verify_certs": false
}
```

File: airflow/providers/opensearch/hooks/opensearch.py (hunk 1 above)
Comment (Lukas1v) [reply airflow/providers/opensearch/hooks/opensearch.py]: I think I have done the requested changes, could you take another look?

**Review by cjames23 — APPROVED** [review]
Looks good to me.

File: airflow/providers/opensearch/hooks/opensearch.py [hunk 2]
```diff
@@ -40,13 +44,20 @@ class OpenSearchHook(BaseHook):
     conn_type = "opensearch"
     hook_name = "OpenSearch Hook"
 
-    def __init__(self, open_search_conn_id: str, log_query: bool, **kwargs: Any):
+    def __init__(
+        self,
+        open_search_conn_id: str,
+        log_query: bool,
+        open_search_conn_class: type[OpenSearchConnectionClass] | None = RequestsHttpConnection,
```
Comment (Taragolis) [inline airflow/providers/opensearch/hooks/opensearch.py:51]: Is it required to change default OpenSearch Connection Type? I guess by default it use `urllib3` (None in this case)

Comment (Lukas1v) [reply airflow/providers/opensearch/hooks/opensearch.py:51]: The Opensearch client indeed uses `urllib3` as default, but the hook actually defaults to `RequestsHttpConnection`. I chose to keep the hook's default to make sure nothing should break.

**Review by eladkal — APPROVED** [review]

**Pull request merged**